src/Raven.Server/Documents/Sharding/Background/ShardedPeriodicDocumentsMigrator.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Raven.Server.Background;
+using Raven.Server.ServerWide.Commands.Sharding;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils;
+using Sparrow.Logging;
+
+namespace Raven.Server.Documents.Sharding.Background
+{
+    public class ShardedPeriodicDocumentsMigrator : BackgroundWorkBase
+    {
+        private readonly ShardedDocumentDatabase _database;
+
+        public ShardedPeriodicDocumentsMigrator(ShardedDocumentDatabase database) : base(database.ShardedDatabaseName, database.ShardedDocumentsStorage.DocumentDatabase.DatabaseShutdown)
+        {
+            _database = database;
+        }
+
+        protected override async Task DoWork()
+        {
+            await WaitOrThrowOperationCanceled(_database.Configuration.Sharding.PeriodicDocumentsMigrationInterval.AsTimeSpan);
+
+            await ExecuteMoveDocuments();
+        }
+
+        internal async Task ExecuteMoveDocuments()
+        {
+            try
+            {
+                var etag = 0L;
+                var buckets = new Dictionary<int, int>();
+
+                using (_database.ShardedDocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                {
+                    while (true)
+                    {
+                        buckets.Clear();
+                        context.Reset();
+                        context.Renew();
+
+                        using (context.OpenReadTransaction())
+                        {
+                            var record = _database.ReadDatabaseRecord();
+                            var documents = _database.ShardedDocumentsStorage.GetDocumentsFrom(context, etag, start: 0, take: 1024).ToList();
```

Comment:
Nope. You cannot do that
This will *continuously* scan through the entire database.
That has *huge* costs in terms of I/O.

Reply:
I think we can use the 'AllDocsBucketAndEtagSlice' index that we have in sharded databases, and just get documents that do not belong to any of our buckets directly?

Reply:
done